### PR TITLE
Add org.yuzu_emu.yuzu

### PR DESCRIPTION
**Pull request by liushuyu**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/yuzu-emu/yuzu-emu.github.io/commits?author=liushuyu
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### Permission Explanations

- Basic Permissions:

```
--socket=x11
--socket=wayland
--socket=pulseaudio
--share=network
--share=ipc
```

- Required for loading ROM files and decryption keys from the host (probably from the console SD card): 

```
filesystem=host:to
```

- Required for using USB game controllers, game controller dongles and webcams/microphones (for emulating cameras and microphones):

```
--device=all
```

### Other Peculiarities

- `only-arches`: Currently only supports `x86_64` architecture, as the CPU JIT recompiler/translator is only usable on this architecture. Also, Nintendo Switch is a very powerful device, and its graphics device can't be emulated through the OpenGL ES API (which is the only available graphics API in Qt for aarch64/arm).

- Use of `io.qt.qtwebengine.BaseApp`: Required for emulating the web browser in the Nintendo Switch. Many games use the web browser for various purposes.

- Same maintainer as `org.citra_emu.citra`: Yeah, these are two _different_ emulator teams sharing the same maintainer.

---
This change is [Reviewable](https://reviewable.io/reviews/flathub/flathub/2139)

### CODE REVIEW

**Comment by liushuyu** [comment]
bot, build org.yuzu_emu.yuzu

File: org.yuzu_emu.yuzu.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  <project_license>GPL-2.0</project_license>
+  <content_rating type="oars-1.0" />
+  <description>
+    <p>yuzu is an experimental open-source emulator for the Nintendo Switch from the creators of Citra.</p>
+    <p>yuzu only emulates a subset of Switch hardware and therefore most commercial games do not run at full speed or are not fully functional. Please visit our Compatibility page (https://yuzu-emu.org/game/) for more information.</p>
+  </description>
+  <launchable type="desktop-id">org.yuzu_emu.yuzu.desktop</launchable>
+  <provides>
+    <binary>yuzu</binary>
+    <id>yuzu.desktop</id>
```
Comment (bilelmoussaoui) [inline org.yuzu_emu.yuzu.metainfo.xml:16]: Does it really provide such app-id?

File: org.yuzu_emu.yuzu.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline org.yuzu_emu.yuzu.metainfo.xml:1]: Should be upstreamed if possible

File: org.yuzu_emu.yuzu.json [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+                "-DBUILD_REPOSITORY=yuzu-emu/yuzu-mainline"
+            ],
+            "cleanup": [
+              "/share/man",
+              "/share/pixmaps"
+            ],
+            "post-install": [
+                "install -Dm644 ../org.yuzu_emu.yuzu.metainfo.xml /app/share/appdata/org.yuzu_emu.yuzu.metainfo.xml",
+                "desktop-file-install --dir=/app/share/applications ../dist/yuzu.desktop",
+                "echo 'StartupWMClass=yuzu' >> /app/share/applications/yuzu.desktop",
```
Comment (bilelmoussaoui) [inline org.yuzu_emu.yuzu.json:183]: desktop-file-edit can be used here, but probably worth a patch upstream.

Comment (liushuyu) [reply org.yuzu_emu.yuzu.json]: Agreed. I will be batching all the suggestions and fixes into one PR (hopefully) on the upstream.

**Comment by liushuyu** [comment]
bot, build org.yuzu_emu.yuzu

**Comment by liushuyu** [comment]
bot, build org.yuzu_emu.yuzu

**Comment by liushuyu** [comment]
I have discovered an issue in the Flatpak version of Yuzu that prevented Yuzu from starting up. My current guess is that the compiler optimization flags were problematic. A patch has been added to test this assumption.

**Comment by liushuyu** [comment]
The build seems stuck. Can anyone restart the build agent?

**Comment by liushuyu** [comment]
> I have discovered an issue in the Flatpak version of Yuzu that prevented Yuzu from starting up. My current guess is that the compiler optimization flags were problematic. A patch has been added to test this assumption.

This is being fixed in https://github.com/yuzu-emu/yuzu/pull/6016.

File: org.yuzu_emu.yuzu.json [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    "runtime-version": "5.15",
+    "sdk": "org.kde.Sdk",
+    "base": "io.qt.qtwebengine.BaseApp",
+    "base-version": "5.15",
+    "command": "yuzu",
+    "rename-desktop-file": "yuzu.desktop",
+    "rename-icon": "yuzu",
+    "finish-args": [
+        "--device=all",
+        "--socket=x11",
```
Comment (barthalion) [inline org.yuzu_emu.yuzu.json:13]:
```suggestion
        "--socket=fallback-x11",
```

**Comment by barthalion** [comment]
I have too new Switch to really test it, so if you consider it ready, it's good to merge after addressing mine and Bilal's comments.

**Comment by liushuyu** [comment]
> I have too new Switch to really test it

You can test it by downloading Switch homebrews. These don't require a Switch to dump/download and should be good enough for a smoke test.

We have tested it and fixed some very obvious Flatpak-specific issues upstream. The patch in the repository will be removed in the next release.

Also, just a side note, if you really want to hack your Switch anyways (risks involved of course), first you need to confirm that your Switch is really patched of the exploit: https://damota.me/ssnc/checker/. If your Switch has been patched but you are skilled enough, you could consider hard-mod your console.

**Comment by liushuyu** [comment]
bot, build org.yuzu_emu.yuzu

**Comment by liushuyu** [comment]
bot, build org.yuzu_emu.yuzu

**Comment by liushuyu** [comment]
bot, build org.yuzu_emu.yuzu

**Comment by liushuyu** [comment]
bot, build org.yuzu_emu.yuzu

**Comment by barthalion** [comment]
LGTM. Anyone else from the upstream project we should invite?

**Comment by LeandroStanger** [comment]
bot, build org.yuzu_emu.yuzu

**Review by LeandroStanger — CHANGES_REQUESTED** [review]

File: org.yuzu_emu.yuzu.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  <url type="contact">https://community.citra-emu.org/</url>
+  <screenshots>
+    <screenshot type="default">https://raw.githubusercontent.com/yuzu-emu/yuzu-emu.github.io/master/images/screenshots/001-Super%20Mario%20Odyssey%20.png</screenshot>
+    <screenshot>https://raw.githubusercontent.com/yuzu-emu/yuzu-emu.github.io/master/images/screenshots/004-Pokemon%20Sword.png</screenshot>
+    <screenshot>https://raw.githubusercontent.com/yuzu-emu/yuzu-emu.github.io/master/images/screenshots/007-Astral%20Chain.png</screenshot>
+    <screenshot>https://raw.githubusercontent.com/yuzu-emu/yuzu-emu.github.io/master/images/screenshots/010-Animal%20Crossing%20New%20Horizons.png</screenshot>
+    <screenshot>https://raw.githubusercontent.com/yuzu-emu/yuzu-emu.github.io/master/images/screenshots/037-The%20Legend%20of%20Zelda%20Links%20Awakening.png</screenshot>
+  </screenshots>
+  <releases>
+    <release date="2021-02-28" version="0-551"/>
```
Comment (LeandroStanger) [inline org.yuzu_emu.yuzu.metainfo.xml:33]:
```suggestion
    <release date="2021-03-05" version="0-555"/>
```

**Review by LeandroStanger — CHANGES_REQUESTED** [review]

**Comment by barthalion** [comment]
ping @liushuyu – I consider it ready to merge, just want to double-check if we should add someone additional to the Flathub repo.

**Comment by liushuyu** [comment]
> @liushuyu Could the flatpak specific issue be resolved? If not, is it possible to follow/support its progress?

Sorry for the delay. Yuzu is focusing on some other more prioritized tasks at the moment (rewriting/refactoring several sub-systems), and hopefully, the rewrites/refactors will eliminate the root cause of those issues.

You can follow our progress here: https://yuzu-emu.org/entry/.

**Comment by liushuyu** [comment]
bot, build org.yuzu_emu.yuzu

**Comment by liushuyu** [comment]
bot, build org.yuzu_emu.yuzu

**Comment by liushuyu** [comment]
Blocked by https://github.com/yuzu-emu/yuzu/pull/6436. Otherwise, it should be good to go.

The previously mentioned issue with Vulkan has been resolved.

**Comment by liushuyu** [comment]
bot, build org.yuzu_emu.yuzu

**Comment by liushuyu** [comment]
bot, build org.yuzu_emu.yuzu

**Comment by liushuyu** [comment]
Thanks!

**Pull request closed without merging**